app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -103,13 +108,29 @@ async function switchEthereumChainHandler(
       }),
     );
   }
-  const requestData = findExistingNetwork(_chainId, findNetworkConfigurationBy);
-  if (requestData) {
+
+  const requestData = {
+    toNetworkConfiguration: findExistingNetwork(
+      _chainId,
+      findNetworkConfigurationBy,
+    ),
+  };
+
+  requestData.fromNetworkConfiguration = getProviderConfig();
+
+  if (requestData.toNetworkConfiguration) {
     const currentChainId = getCurrentChainId();
+
+    // we might want to change all this so that it displays the network you are switching from -> to (in a way that is domain - specific)
+
+    const networkClientId = findNetworkClientIdByChainId(_chainId);
```

Comment:
we still need to figure out a solution for when there are multiple networkClients for a chainId, but this is true of today so obviously not a requirement here. Just noting.